src/components/wallet/ActivitiesWrapper.js

```diff
@@ -84,28 +86,36 @@ const StyledContainer = styled.div`
     }
 `;
 
-const Activities = ({ transactions, accountId, getTransactionStatus }) => {
+const ActivitiesWrapper = () => {
+    const dispatch = useDispatch();
+
     const [transactionHash, setTransactionHash] = useState();
     const activityLoader = actionsPending(['GET_TRANSACTIONS', 'REFRESH_ACCOUNT_OWNER']);
-    const transaction = useSelector((state) => selectOneTransactionByHash(state, { accountId, hash: transactionHash }));
+
+    const accountId = useSelector(state => selectAccountId(state));
+    const transactions = useSelector(state => selectTransactionsByAccountId(state, { accountId }));
+    const transaction = useSelector(state => selectOneTransactionByHash(state, { accountId, hash: transactionHash }));
+
+    useEffect(() => {
+        if (accountId) {
+            dispatch(getTransactions(accountId));
+        }
+    }, [accountId]);
 
     return (
         <StyledContainer>
             <h2 className={classNames({'dots': activityLoader})}><Translate id='dashboard.activity' /></h2>
-            {transactions
-                ? transactions.map((transaction, i) => (
-                    <ActivityBox
-                        key={i}
-                        transaction={transaction}
-                        actionArgs={transaction.args}
-                        actionKind={transaction.kind}
-                        receiverId={transaction.receiver_id}
-                        accountId={accountId}
-                        setTransactionHash={setTransactionHash}
-                    />
-                ))
-                : null
-            }
+            {transactions.map((transaction, i) => (
+                <ActivityBox
+                    key={i}
```

Comment:
`key` here should be the transaction hash so they are unique for correctness; using index for key is an anti-pattern in React :) (ref: https://robinpokorny.com/blog/index-as-a-key-is-an-anti-pattern/)

Reply:
Right, didn't notice this bug while refactoring. Done.